src/analysis.jl

```diff
@@ -25,47 +25,45 @@ function code(c::DynamicCall; optimize = false)
   codeinfo = code_typed(c.f, argtypes(c), optimize = optimize)
   @assert length(codeinfo) == 1
   codeinfo = codeinfo[1]
-  linearize!(codeinfo[1])
   return codeinfo
 end
 
-struct StaticCall <: Call
-  method_instance::MethodInstance
-end
-
-argtypes(c::StaticCall) = Tuple{c.method_instance.specTypes.parameters[2:end]...}
-types(c::StaticCall) = c.method_instance.specTypes
-method(c::StaticCall) = c.method_instance.def
-
-method_expr(c::StaticCall) = method_expr(method(c).name, argtypes(c))
-
-function code(c::StaticCall; optimize = false)
-  # TODO static call graph can only be computed with optimize=true, so analyzing with optimized=false will skip inlined methods
-  codeinfo = get_code_info(c.method_instance, optimize=true)
-  linearize!(codeinfo[1])
-  return codeinfo
-end
+# struct StaticCall <: Call
+#   method_instance::MethodInstance
+# end
+#
+# argtypes(c::StaticCall) = Tuple{c.method_instance.specTypes.parameters[2:end]...}
+# types(c::StaticCall) = c.method_instance.specTypes
+# method(c::StaticCall) = c.method_instance.def
+#
+# method_expr(c::StaticCall) = method_expr(method(c).name, argtypes(c))
+#
+# function code(c::StaticCall; optimize = false)
+#   # TODO static call graph can only be computed with optimize=true, so analyzing with optimized=false will skip inlined methods
+#   codeinfo = get_code_info(c.method_instance, optimize=true)
+#   linearize!(codeinfo[1])
+#   return codeinfo
+# end
 
 function eachline(f, code, line = -1)
-  for l in code.code
-    if l isa LineNumberNode
-      line = l.line
-    else
-      f(line, l)
-    end
+  for (i, l) in enumerate(code.code)
+    ind = code.codelocs[i]
+    ind = clamp(ind, 1, length(code.linetable))
```

Comment:
Perhaps we should just set the line to `nothing` in this case, otherwise it could be misleading

Reply:
Or `-1`, which would be in line with how Base handles that.

Reply:
Yeah that would be fine, as long as we do a reasonable printing.